encrypted-media-respec.html

```diff
@@ -638,42 +691,83 @@
                 "optional"    ||       "required"       |      "not-allowed"     | Depends on combination |
                 "not-allowed" ||       Return null      |      "not-allowed"     |      "not-allowed"     |
                -->
-              <li><p>Follow the steps for the value of <var>candidate configuration</var>'s <a def-id="option-distinctiveIdentifier"></a> member from the following list:</p>
+              <li>
+                <p>
+                  Let <var>distinctive identifier requirement</var> be the value of <var>candidate configuration</var>'s <a def-id="option-distinctiveIdentifier"></a> member.
+                </p>
+              </li>
+              <li>
+                <p>
+                  If <var>distinctive identifier requirement</var> is <a def-id="requirement-optional"></a> and
+                  <a def-id="distinctive-identifiers"></a> are not allowed according to <var>restrictions</var>, set <var>distinctive identifier requirement</var>
+                  to <a def-id="requirement-not-allowed"></a>.
+                </p>
+              <li><p>Follow the steps for <var>distinctive identifier requirement</var> from the following list:</p>
                 <dl class="switch">
                   <dt><a def-id="requirement-required"></a></dt>
                   <dd>
-                    <p>If the <var>implementation</var> does not support a persistent <a def-id="distinctive-identifier"></a> in combination with <var>accumulated configuration</var>, return <code>null</code>.</p>
+                    <p>If the <var>implementation</var> does not support a persistent <a def-id="distinctive-identifier"></a> in combination with <var>accumulated configuration</var> and <var>restrictions</var>, return <code>NotSupported</code>.</p>
                   </dd>
                   <dt><a def-id="requirement-optional"></a></dt>
                   <dd>
-                    <p>Continue.<!-- Will be updated below. --></p>
+                    <p>Continue with the following steps.</p>
                   </dd>
                   <dt><a def-id="requirement-not-allowed"></a></dt>
                   <dd>
-                    <p>If the <var>implementation</var> requires a <a def-id="distinctive-identifier"></a> in combination with <var>accumulated configuration</var>, return <code>null</code>.</p>
+                    <p>If the <var>implementation</var> requires a <a def-id="distinctive-identifier"></a> in combination with <var>accumulated configuration</var> and <var>restrictions</var>, return <code>NotSupported</code>.</p>
```

Comment:
Related to the next comment:
Looking at the (existing and new) text more closely, it may not be clear/correct. "A requires X in combination with B and C." I believe it is A, B, and C that are the combination and X is the result. Or A requires X when B and C. Either way, "in combination with" seems kludgy.
One option would be to worded these as "If X is required by A when in the configuration B with restrictions C applied." It's a bit wordy, though. However, this _might_ eliminate the need for the note below.

This should probably be addressed separately since it already exists and is unrelated to these changes, though they exacerbate the issue.